presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -235,8 +236,14 @@ private static void runConsole(QueryRunner queryRunner, ClientSession session, A
 
                         process(queryRunner, split.statement(), outputFormat, true);
                     }
-                    reader.getHistory().add(squeezeStatement(split.statement()) + split.terminator());
+                    if (statements.length() == 0) {
+                        statements.append(squeezeStatement(split.statement())).append(split.terminator());
+                    }
+                    else {
+                        statements.append(" ").append(squeezeStatement(split.statement())).append(split.terminator());
+                    }
                 }
+                reader.getHistory().add(statements);
```

Comment:
This needs to check if statements is empty. Otherwise, you append a blank line to the history when there are no complete statements:
```SQL
SELECT 1,
2,
3;